docs/dev/stream/state/custom_serialization.md

```diff
@@ -223,15 +227,38 @@ the same `TypeSerializerSnapshot` class as their snapshot would complicate the i
 This would also be a bad separation of concerns; a single serializer's serialization schema,
 configuration, as well as how to restore it, should be consolidated in its own dedicated `TypeSerializerSnapshot` class.
 
-#### 3. Use the `CompositeSerializerSnapshot` utility for serializers that contain nested serializers
+### Extending `CompositeTypeSerializerSnapshot` for serializer snapshot with nested serializers
 
-There may be cases where a `TypeSerializer` relies on other nested `TypeSerializer`s; take for example Flink's
-`TupleSerializer`, where it is configured with nested `TypeSerializer`s for the tuple fields. In this case,
-the snapshot of the most outer serializer should also contain snapshots of the nested serializers.
+Before further explanation, we call the serializer, which relies on other nested serializer(s), as "outer" serializer in this context.
+Eamples for this could be `MapSerializer`, `ListSerializer`, `GenericArraySerializer`, etc., and take `MapSerializer` for example,
+the map-key and map-value serializers consist as the nested serializers, while `MapSerialize` itself is the "outer" serializer.
+In this case, the snapshot of the most outer serializer should also contain snapshots of the nested serializers.
+Besides, unlike the former two serializers, `GenericArraySerializer` also contains some extra static information, a class of the component type,
+that needs to be persisted beyond the nested component serializer.
 
-The `CompositeSerializerSnapshot` can be used specifically for this scenario. It wraps the logic of resolving
+`CompositeTypeSerializerSnapshot` is a convenient serializer snapshot class for serializers which delegate its serialization
+to multiple nested serializers, and also useful for serializers which need to persist some extra static information as its snapshot.
```

Comment:
`CompositeTypeSerializerSnapshot` is designed to assist in the implementation of snapshotting for serializers which delegate their serialization
to multiple nested serializers. It's also helpful in cases where some extra static information needs to be persisted.  `CompositeTypeSerializerSnapshot` takes care of the overall schema compatibility check for the composite serializer.